src/azure-openapi-validator/rules/DescriptionMustNotBeNodeName.ts

```diff
@@ -0,0 +1,34 @@
+import { rules } from "../rule";
+
+rules.push({
+  id: "M2065",
+  name: "DescriptionMustNotBeNodeName",
+  severity: "error",
+  category: ["RPCViolation"],
+
+  appliesTo_JsonQuery: "$..*[?(@.description)]",
+  run: function* (doc, node, path) {
+    if (node.description === undefined) {
```

Comment:
as explained, this is guaranteed to not be the case due to your query.
However, you should check whether `node.description` is a string since it could well be something else, e.g. a property of a model type.